agent/agent.go

```diff
@@ -379,6 +335,93 @@ func (a *Agent) loop(ctx context.Context) error {
 	}
 }
 
+// ProcessNextResponse contains information for what should be done after processing next task
+type ProcessNextResponse struct {
+	ShouldExit       bool
+	ShouldContinue   bool
+	ShouldResetTimer bool
+	needPostGroup    bool
+	tc               *taskContext
+}
+
+// ProcessNextTask takes in the NextTaskResponse and processes it.
+func (a *Agent) ProcessNextTask(nt *apimodels.NextTaskResponse, ctx context.Context, tc *taskContext, needPostGroup bool) (ProcessNextResponse, error) {
+	if nt.ShouldExit {
+		grip.Notice("Next task response indicates agent should exit.")
+		return ProcessNextResponse{ShouldExit: true}, nil
+	}
+	// if the host's current task group is finished we teardown
+	if nt.ShouldTeardownGroup {
+		a.runPostGroupCommands(ctx, tc)
+		return ProcessNextResponse{
+			ShouldContinue:   true,
```

Comment:
Rather than pass back `ShouldContinue` and `ShouldResetTimer` as identical values in basically every single case, it seems like we could capture a single field meaning "the agent should sleep again" (i.e. a `noTaskToRun` field or something to that effect) and make that the default rather than "the agent should continue". This would be cleaner and more meaningful for the reader because it makes it clear that having no new task to run means we sleep for a little while, waiting for another task. It also makes the default case better then the default case is to reset the timer to 0 (and ask for another task), and only in the one special case (when there's no new task to run) do you choose to sleep longer. Something like this:
```go
ntr, err := processNextTask(...)
// error handling and whatever
if ntr.noTaskToRun {
    // Increase sleep interval and sleep until next round of asking for a task
    continue
}

// Happy path - we ran something, so we should reset the timer and immediately ask for a new task
```